include/gadget/usdt_argument.h

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2024 The Inspektor Gadget authors
+
+#ifndef __USDT_ARGUMENT_H
+#define __USDT_ARGUMENT_H
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__type(key, __u32);
+	__type(value, __u32);
+	__uint(max_entries, 1);
+} __usdt_args_placeholder SEC(".maps");
+
+/* Returns NULL pointer on failure, otherwise will return an u64* pointer */
+/* The buffer will be reused, so users need to store the value before next call */
+/* Interface only here, needs to be replaced by bpf extension */
+__attribute__((noinline)) u64 *usdt_get_argument(struct pt_regs *ctx,
+						 int arg_idx)
```

Comment:
This function will be replaced at runtime with `BPF_F_REPLACE` by "freplace/usdt_get_argument" defined in usdt_helper.bpf.c below, so `arg_idx` will be used.

The code below `volatile u64 *ret = ctx;` is therefore never executed. But it is necessary, otherwise the compiler will think the function is a noop (the compiler does not know we will replace the function at runtime by a bpf extension with `BPF_F_REPLACE`) and calls to the function might be optimized out. By using the `volatile`, we force the compiler to skip that optimization.